src/textual/dom.py

```diff
@@ -279,6 +282,25 @@ def get_path(base: Type[DOMNode]) -> str:
 
         return css_stack
 
+    def get_component_classes(self) -> set[str]:
```

Comment:
Any reason why this should be public?

Unless there is a good reason to expose this to the dev, we should make it "private".

Reply:
I made it public instead of private because `Stylesheet.apply` needs it and because this feels analogous to `get_default_css`, which is public.

Reply:
If this is staying "public", it might make sense to reword the docstring so that it doesn't refer to `_inherit_component_classes`.

Reply:
We use a more relaxed definition of public and private. Essentially `public` is something that the Textual dev may need to use. If it is only used by the Textual system then it should be `private`.

A better term, might be `internal`.

Reply:
I reworded the docstring, good catch!

I don't think the average Textual dev will need to use `get_component_classes` but I did this in light of `get_default_css` because it is more or less the same level of thing.
Should I prefix both with an underscore `_`?

Reply:
Sure. May as well.